lib/routemaster/tasks/fix_cache_ttl.rb

```diff
@@ -0,0 +1,45 @@
+require 'routemaster/cache_key'
+
+module Routemaster
+  module Tasks
+    class FixCacheTTL
+      def initialize(cache: Config.cache_redis, batch_size: 100)
+        @cache = cache
+        @batch_size = batch_size
+      end
+
+      def call
+        pattern = "#{@cache.namespace}:#{CacheKey::PREFIX}*"
+        _each_key_batch(pattern) do |node, keys|
+          _fix_keys(node, keys)
+        end
+      end
+
+      private
+
+      def _each_key_batch(pattern)
+        @cache.redis.nodes.each do |node|
+          cursor = 0
+          loop do
+            cursor, keys = node.scan(cursor, count: @batch_size, match: pattern)
+            yield node, keys
+            break if cursor.to_i == 0
+          end
+        end
+      end
+
+      def _fix_keys(node, keys)
+        ttls = node.pipelined do |p|
+          keys.each { |k| p.ttl(k) }
+        end
+
+        node.pipelined do |p|
+          keys.zip(ttls).each do |k,ttl|
+            next unless ttl < 0
+            node.expire(k, Config.cache_expiry)
```

Comment:
If we have thousands and thousands of keys all expiring at the same time, will it have any impact on performance? Or because they will be evicted over time, the impact is limited?

Reply:
The latter — `Config.cache_expiry` should be 1 year in the future, so eviction is most likely to happen due to memory constraints, earlier, rather than at the expiry timestamp.

Reply:
I think we changed this to 90 days relatively recently.